lib/manageiq_foreman/lib/manageiq_foreman/connection.rb

```diff
@@ -0,0 +1,119 @@
+module ManageiqForeman
+  class Connection
+    attr_accessor :connection_attrs
+
+    def initialize(connection_attrs)
+      @connection_attrs = connection_attrs
+    end
+
+    def api_version
+      raw_home.status.first["api_version"]
+    end
+
+    def all(method, filter = {})
+      expected_size = nil
+      page = 0
+      all = []
+
+      while expected_size.nil? || all.size < expected_size
+        small = public_send(method, filter.merge(:page => (page += 1), :per_page => 50))
+        expected_size ||= small.total
+        break if small.empty?
+        all += small.to_a
+      end
+      PagedResponse.new(all)
```

Comment:
You have a paged_response method down below, but don't use it?  Also why the need for that method...`PagedResponse.new(all)` is just as clear as `paged_response(all)`